### PR TITLE
[ARCHETYPE-658] Upgrade Groovy from 2.x to 4.0.22

### PR DESCRIPTION
**Pull request by slachiewicz**

### CODE REVIEW

File: maven-archetype-plugin/src/main/java/org/apache/maven/archetype/mojos/IntegrationTestMojo.java [hunk 1]
```diff
@@ -698,7 +691,7 @@ private FileLogger setupLogger(File basedir) throws IOException {
             File outputLog = new File(basedir, "build.log");
 
             if (streamLogs) {
-                logger = new FileLogger(outputLog, getLog());
+                logger = new FileLogger(outputLog, LoggerFactory.getLogger(getClass()));
```
Comment (slachiewicz) [inline maven-archetype-plugin/src/main/java/org/apache/maven/archetype/mojos/IntegrationTestMojo.java:694]: not sure here

File: maven-archetype-plugin/src/main/java/org/apache/maven/archetype/mojos/IntegrationTestMojo.java [hunk 2]
```diff
@@ -671,22 +672,14 @@ private void invokePostArchetypeGenerationGoals(String goals, File basedir, File
...
-            scriptRunner.run(
-                    "post-build script",
-                    goalFile.getParentFile(),
-                    postBuildHookScript,
-                    context,
-                    logger,
-                    "failure post script",
-                    true);
-        } catch (RunFailureException e) {
+            scriptRunner.run("post-build script", goalFile.getParentFile(), postBuildHookScript, context, logger);
```
Comment (slachiewicz) [inline maven-archetype-plugin/src/main/java/org/apache/maven/archetype/mojos/IntegrationTestMojo.java:681]: not sure here

Comment (slawekjaranowski) [reply maven-archetype-plugin/src/main/java/org/apache/maven/archetype/mojos/IntegrationTestMojo.java:681]: looks ok

File: maven-archetype-plugin/src/main/java/org/apache/maven/archetype/mojos/IntegrationTestMojo.java (hunk 1 above)
Comment (slawekjaranowski) [reply maven-archetype-plugin/src/main/java/org/apache/maven/archetype/mojos/IntegrationTestMojo.java:694]: looks ok

**Review by slawekjaranowski — APPROVED** [review]
I hope we can go with it

**Comment by slawekjaranowski** [comment]
@slachiewicz can we go with it?

**Review by slawekjaranowski — APPROVED** [review]

**Comment by jira-importer** [comment]
Resolve #888

**Pull request merged**